src/main/java/io/pinecone/clients/Pinecone.java

```diff
@@ -871,6 +871,10 @@ public AsyncIndex getAsyncIndexConnection(String indexName) throws PineconeValid
         return new AsyncIndex(connection, indexName);
     }
 
+    public Inference getInference() {
+        return new Inference();
+    }
```

Comment:
How heavy is it to create this object, or should we cache Inference in Pinecone? 

e.g. if someone effectively did this every time:

```java
pc.getInference().embed(...);
pc.getInference().embed(...);
pc.getInference().embed(...);
pc.getInference().embed(...);
```

Reply:
Would you please elaborate on this, I didn't understand your question here. Are you concerned that if the user, instead of using the same object, instantiates a new object for Inference class and will result in multiple objects in heap memory instead of referencing to one single object in the heap memory?

Reply:
Yeah I see your point! Do you feel like java users are traditionally used to getting the Inference object and reusing it rather than instantiating every single time to call the embed()?

Reply:
I don't think there's a java convention for this, so just thinking about the ergonomics of how we deliver the API to them, and if there's something here that's heavy to construct/connect/etc. then we should try to lazy construct that and reuse across requests

Reply:
Maybe calling it `getInferenceClient()` would resolve the ambiguity and encourage use like:

```java
client = pc.getInferenceClient();
client.embed(...);
client.embed(...);
```

Reply:
I'm not worried about the extra objects if they're lightweight to construct. So I'm asking  how expensive are they to construct? 

> How heavy is it to create this object, or should we cache Inference in Pinecone?